### PR TITLE
Issue/2071

### PR DESCRIPTION
**Pull request by mehul0810**
## Description
This PR is for #2071 

## How Has This Been Tested?
I have tested this with various scenarios:
1. Simple Donation with PayPal
2. Multi Level Donation with PayPal
3. Simple Donation with Custom Amount
4. Multi Level Donation with various levels
5. Multi Level Donation with Custom Amount
6. Simple Donation with Fee Recovery
7. Simple Donation with Fee Recovery and Custom Amount
8. Multi Level Donation with Fee Recovery
9. Multi Level Donation with Fee Recovery and Custom Amount.

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: includes/gateways/paypal-standard.php [hunk 1]
```diff
@@ -693,8 +695,9 @@ function give_get_paypal_button_type() {
  * @return string
  */
 function give_build_paypal_item_title( $payment_data ) {
```
Comment (DevinWalker) [inline includes/gateways/paypal-standard.php]: @mehul0810 this function will be useful outside of the PayPal gateway. Rename it to `give_payment_gateway_item_title()` and place it above https://github.com/WordImpress/Give/blob/master/includes/misc-functions.php#L395

Comment (DevinWalker) [inline includes/gateways/paypal-standard.php]: Add a filter for the return output named `give_payment_gateway_item_title` and also pass the `$form_id` and `$payment_data` as params.

Comment (mehul0810) [reply includes/gateways/paypal-standard.php]: @DevinWalker Moved function to `misc-functions.php` above line no 395

Comment (mehul0810) [reply includes/gateways/paypal-standard.php]: @DevinWalker Added Filter `give_payment_gateway_item_title` with proper comments

File: includes/misc-functions.php [hunk 2]
```diff
@@ -380,6 +380,57 @@ function give_get_purchase_session() {
...
 /**
+ * Build item title for PayPal.
+ *
+ * @param array $payment_data Payment Data.
+ *
+ * @since 1.8
+ *
+ * @return string
+ */
+function give_build_paypal_item_title( $payment_data ) {
```
Comment (DevinWalker) [inline includes/misc-functions.php]: You didn't update the function name like I asked

File: includes/misc-functions.php [hunk 3]
```diff
@@ -380,6 +380,57 @@ function give_get_purchase_session() {
 }
 
 /**
+ * Build item title for PayPal.
+ *
+ * @param array $payment_data Payment Data.
+ *
+ * @since 1.8
```
Comment (DevinWalker) [inline includes/misc-functions.php]: Incorrect version

File: includes/misc-functions.php [hunk 4]
```diff
@@ -380,15 +380,15 @@ function give_get_purchase_session() {
...
  *
  * @param array $payment_data Payment Data.
  *
- * @since 1.8
+ * @since 1.8.14
  *
  * @return string
  */
-function give_build_paypal_item_title( $payment_data ) {
+function give_payment_gateway_item_title( $payment_data ) {
```
Comment (DevinWalker) [inline includes/misc-functions.php]: @mehul0810 can you write a unit test for this function so we're certain it works through the test conditions you listed?

Comment (mehul0810) [reply includes/misc-functions.php]: Sure @DevinWalker . I'll write unit tests and will update you.

Comment (DevinWalker) [reply includes/misc-functions.php]: @mehul0810 any luck?

Comment (mehul0810) [reply includes/misc-functions.php]: @DevinWalker Yes... I have added unit test for simple donation forms. But, working on tests for Multi Level Donation for function `give_payment_gateway_item_title`. Updating you in some time.

Comment (mehul0810) [reply includes/misc-functions.php]: @DevinWalker I've added unit tests and make sure that it is working properly.

File: includes/misc-functions.php (hunk 2 above)
Comment (mehul0810) [reply includes/misc-functions.php]: @DevinWalker Function Updated.

File: tests/framework/helpers/class-helper-form.php [hunk 5]
```diff
@@ -174,6 +174,112 @@ public static function create_form( $args ) {
...
+	 * Setup Simple Donation Form with Post Data.
+	 *
+	 * @param bool $is_custom_amount Status for custom amount enabled or not.
+	 *
+	 * @since 1.8.14
+	 * @access public
+	 *
+	 * @return array
+	 */
+	public static function setup_simple_donation_form( $is_custom_amount = false ) {
```
Comment (DevinWalker) [inline tests/framework/helpers/class-helper-form.php]: @mehul0810 are you using this as a helper way to create a simple form?

File: includes/deprecated/deprecated-functions.php [hunk 6]
```diff
@@ -623,3 +623,20 @@ function give_get_purchase_summary( $purchase_data, $email = true ) {
...
+ *
+ * @since 1.8.14
+ */
+function give_build_paypal_item_title( $payment_data ) {
+
+	$backtrace = debug_backtrace();
+
+	_give_deprecated_function( __FUNCTION__, '1.8.14', 'give_payment_gateway_item_title', $backtrace );
+
+	give_payment_gateway_item_title( $payment_data );
```
Comment (ravinderk) [inline includes/deprecated/deprecated-functions.php]: @mehul0810 this will return value, please fix it

**Pull request merged**